### ISSUE TITLE
Permission denied on start

### ISSUE THREAD

**Issue by Moulde** [cdalvaro/docker-salt-master#226]
**Describe the bug**
When i try to start the container using the docker-compose file, I get permission denied errors
I also tried starting the container with this command, but same output.

```
docker run --name salt_stack -it --rm \
    --publish 4505:4505 --publish 4506:4506 \
    --env "PUID=$(id -u)" --env "PGID=$(id -g)" \
    --volume $(pwd)/roots/:/home/salt/data/srv/ \
    --volume $(pwd)/keys/:/home/salt/data/keys/ \
    ghcr.io/cdalvaro/docker-salt-master:latest
```

**To Reproduce**
Clone repo and do `docker-compose up`

**Expected behavior**
The container should generate keys etc and start.

**Additional info**
```
[+] Building 0.0s (0/0)                                                                                      
[+] Running 1/0
 ✔ Container salt_master  Created                                                                       0.0s 
Attaching to salt_master
salt_master  | [INFO] - Configuring directories ...
salt_master  | '/srv' -> '/home/salt/data/srv'
salt_master  | '/var/log/salt' -> '/home/salt/data/logs/salt'
salt_master  | [INFO] - Configuring logrotate ...
salt_master  | [INFO] - Configuring container timezone ...
salt_master  | [INFO] - Setting TimeZone -> Europe/Madrid ...
salt_master  | [INFO] - Configuring salt-master service ...
salt_master  | [INFO] - Configuring 3rd-party salt-formulas ...
salt_master  | [INFO] - Setting up salt keys ...
salt_master  | [INFO] -  ==> Setting up master keys ...
salt_master  | [INFO] - Generating master keys ...
salt_master  | Error: [Errno 13] Permission denied: '/home/salt/data/keys/master.pem'
salt_master  | [INFO] -  ==> Setting up GPG keys ...
salt_master  | [INFO] - Could not find GPG keys. GPG setup skipped.
salt_master  | [INFO] - Setting up salt keys permissions ...
salt_master  | chmod: changing permissions of '/home/salt/data/keys/master.pem': Permission denied
salt_master exited with code 1
salt_master  | chown: changing ownership of '/home/salt/data/keys/master.pem': Permission denied
salt_master exited with code 1
salt_master  | chown: changing ownership of '/home/salt/data/keys/master.pem': Permission denied
salt_master exited with code 1
salt_master  | chown: changing ownership of '/home/salt/data/keys/master.pem': Permission denied
salt_master exited with code 1
salt_master  | chown: changing ownership of '/home/salt/data/keys/master.pem': Permission denied
```

**Version report (please complete the following information):**

- Host OS: [e.g. `uname -a`]
`Darwin MBP 23.2.0 Darwin Kernel Version 23.2.0: Wed Nov 15 21:55:06 PST 2023; root:xnu-10002.61.3~2/RELEASE_ARM64_T6020 arm64`
- Docker: [e.g. `docker --version`]
`Docker version 25.0.2, build 29cf629`
- Image tag: [e.g. `3006.6`]
`ghcr.io/cdalvaro/docker-salt-master:latest`

**Additional context**

**Comment by cdalvaro** [comment]
Hi, @Moulde! Thank you for opening this issue!

First of all, check that your `keys` directory has the right permissions:

```sh
sudo chown $(id -u):$(id -g) keys
chmod 644 keys
```

If you still having issues after this, tell me and I'll check it more carefully!

**Comment by cdalvaro** [comment]
> Hi, @Moulde! Thank you for opening this issue!
> 
> First of all, check that your `keys` directory has the right permissions:
> 
> ```shell
> sudo chown $(id -u):$(id -g) keys
> chmod 644 keys
> ```
> 
> If you still having issues after this, tell me and I'll check it more carefully!

Hi @Moulde! Did that work??

**Comment by cdalvaro** [comment]
This seems to be related with: https://github.com/saltstack/salt/issues/65093

But it only happens on macOS.

**Comment by cdalvaro** [comment]
Hi @Moulde, this issue should be fixed in image `3006.7_1`.

Please let me know if the issue persists.

And thank you for opening this issue!
